### PR TITLE
start-game use case

### PR DESCRIPTION
**Pull request by Dylanflini**

### CODE REVIEW

File: minesweeper/use-cases/startGame/startGame.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	for (let y = 0; y < columns; y++) {
+		for (let x = 0; x < rows; x++) {
+			cells[total].position = [x, y]
+			total++
+		}
+	}
+
+	const neighBorsBombsCounter: NeighborsBombsCounter[] = Array(totalCells)
+
+	let bombs: BombModel[] = Array(bombsInput)
```
Comment (danielglobant) [inline minesweeper/use-cases/startGame/startGame.ts:54]: i think this can be const

File: minesweeper/use-cases/startGame/startGame.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * start game base on initial props
+ */
+
+export const startGameUseCase: IStartGameUseCase = ({
+	bombs: bombsInput,
+	rows,
+	columns,
+}) => {
+	if (bombsInput < 1) {
+		throw new Error('bombs must be greater than 0')
```
Comment (danielglobant) [inline minesweeper/use-cases/startGame/startGame.ts:27]: I would put that errors in an Enum

enum Error {
  BOMBS_GREATER_THAN_ZERO = ''bombs must be greater than 0'',
  // each case
}

File: minesweeper/use-cases/startGame/startGame.spec.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	})
+
+	it('should all cells have diferente position', () => {
+		const props = { bombs: 2, columns: 3, rows: 3 }
+
+		const { cells } = startGameUseCase(props)
+	})
+
+	it.todo(
+		'las bombas que se generan no deberian tener la misma posicion que otra'
```
Comment (danielglobant) [inline minesweeper/use-cases/startGame/startGame.spec.ts:87]: The bombs shouldn't have the same position that other bombs

File: minesweeper/use-cases/startGame/startGame.spec.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import { startGameUseCase } from './startGame'
+
+describe('start game', () => {
+	describe('props validations', () => {
+		it('should throw if props columns or rows are not valid', () => {
+			expect(() =>
+				startGameUseCase({ bombs: 1, columns: 3, rows: 1 })
+			).toThrowError('rows must be greater than 1')
```
Comment (danielglobant) [inline minesweeper/use-cases/startGame/startGame.spec.ts:8]: Use enum to create

File: minesweeper/use-cases/startGame/startGame.spec.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+			expect(() =>
+				startGameUseCase({ bombs: 1, columns: 1, rows: 3 })
+			).toThrowError('columns must be greater than 1')
+		})
+
+		it('should throw if props not contain bombs', () => {
```
Comment (danielglobant) [inline minesweeper/use-cases/startGame/startGame.spec.ts:15]: throw errorr

File: minesweeper/use-cases/startGame/startGame.spec.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+			const props = { bombs: -1, rows: 2, columns: 2 }
+
+			expect(() => startGameUseCase(props)).toThrowError(
+				'bombs must be greater than 0'
+			)
+		})
+	})
+
+	it('shoult return initial board', () => {
```
Comment (danielglobant) [inline minesweeper/use-cases/startGame/startGame.spec.ts:24]: shoult -> should

File: minesweeper/use-cases/startGame/startGame.spec.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+		const props = { bombs: 2, columns: 3, rows: 3 }
+
+		const { cells } = startGameUseCase(props)
+	})
+
+	it.todo(
+		'las bombas que se generan no deberian tener la misma posicion que otra'
+	)
+
+	it.todo('should all vecinos have value between 0 to 8')
```
Comment (danielglobant) [inline minesweeper/use-cases/startGame/startGame.spec.ts:90]: neighbors

File: minesweeper/use-cases/startGame/startGame.ts (hunk 1 above)
Comment (Dylanflini) [reply minesweeper/use-cases/startGame/startGame.ts]: i think to be let for push new bomb in this array, you can see new implementation

File: minesweeper/use-cases/startGame/startGame.ts (hunk 2 above)
Comment (Dylanflini) [reply minesweeper/use-cases/startGame/startGame.ts:29]: I agree

File: minesweeper/use-cases/startGame/startGame.ts (hunk 1 above)
Comment (danielglobant) [reply minesweeper/use-cases/startGame/startGame.ts]: Ok but for pushing you can do it with a push or even by index and it doesn't affect to be a const

You can do ` array[5] = false  `in a `const` and it allows you to do that

Comment (Dylanflini) [reply minesweeper/use-cases/startGame/startGame.ts]: It's true 😄

File: minesweeper/use-cases/startGame/startGame.spec.ts (hunk 3 above)
Comment (Dylanflini) [reply minesweeper/use-cases/startGame/startGame.spec.ts]: done

File: minesweeper/use-cases/startGame/startGame.spec.ts (hunk 5 above)
Comment (Dylanflini) [reply minesweeper/use-cases/startGame/startGame.spec.ts]: 👌

**Review by danielglobant — APPROVED** [review]

**Pull request merged**